### PR TITLE
feat: prevent backward navigation when clicking a link in a webview (EMI-1395)

### PR DESCRIPTION
**Pull request by iskounen**
This PR resolves [EMI-1395]

### Description

When users tap on external links (such as help center articles) in the checkout flow, and then return to the checkout flow by tapping the back button, they are navigated to the previous step of the checkout flow. This is a poor user experience.

This PR addresses that by ensuring that the WebView's backward navigation event (`goBack`) isn't called for taps on external links.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

<details><summary>Changelog updates</summary>

#### iOS user-facing changes

- Opening help center links during checkout won't return users to the previous step

</details>

[EMI-1395]: https://artsyproduct.atlassian.net/browse/EMI-1395?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- iOS user-facing changes (Opening help center links during checkout won't return users to the previous step)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 558af8a1b6fa3563f3ca9142083c927799f531eb
</p>

**Comment by iskounen** [comment]
@damassi I've asked for your review on this PR because it makes some changes to logic that you added in #7759 to make sure that routing happens correctly for artwork pages. We believe that logic was causing users to be navigated back one step in the checkout flow when they clicked on help center links.

We removed some debouncing and `!_TEST_` guards because we weren't sure that they were necessary. Please let us know if you have thoughts about that change in particular.

**Comment by damassi** [comment]
👀

**Review by MrSltun — DISMISSED** [review]
Looks great!! 🌟🌟🌟

File: src/app/Components/ArtsyWebView.tests.tsx [hunk 1]
```diff
@@ -220,6 +254,63 @@ describe("ArtsyWebViewPage", () => {
       })
       expect(navigate).toHaveBeenCalledWith("https://google.com")
     })
+
+    describe("the inner WebView's goBack method", () => {
+      it("is called when the URL matches a ReactWebView, ModalWebView, or VanityURLEntity route", () => {
```
Comment (brainbicycle) [inline src/app/Components/ArtsyWebView.tests.tsx:259]: confused about this test case, don't the other test cases contradict it? 

e.g: `it("is called when the URL matches a ReactWebView, ModalWebView, or VanityURLEntity route"` but `it("is not called when the URL matches a ModalWebView route"`

Comment (brainbicycle) [reply src/app/Components/ArtsyWebView.tests.tsx:259]: should it be something like `"it("is not called when the URL matches a ModalWebView route to an external url"` and `it("is called when the URL matches a ReactWebView, ModalWebView, or VanityURLEntity route to a matching route"`?

**Review by gkartalis — APPROVED** [review]
We did some QA during mobile OH and it looks good to go!

File: src/app/Components/ArtsyWebView.tests.tsx (hunk 1 above)
Comment (brainbicycle) [reply src/app/Components/ArtsyWebView.tests.tsx]: 👍 makes sense, thanks for the explanation!

**Review by brainbicycle — APPROVED** [review]

**Pull request merged**